processor/processor_test.go

```diff
@@ -0,0 +1,71 @@
+package processor
```

Comment:
It's better to do black-box testing. To achieve it, use `package processor_test`.

Reply:
What is the benefit to black box testing vs just using the public methods?

One pattern I like is having private helpers (i.e. for dependency injection) to make testing easier, while having simple public functions.

Reply:
In general, white box tests are bad because they are fragile. When internal implementation of a unit changes, the tests also need to change. However, if the tests only invoke the public API, which rarely changes, changes in code don't break the tests.